spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -61,7 +62,8 @@
  * <p>The above list is in order of priority. For example: a matching catalog will take priority
  * over any namespace resolution.
  */
-public class IcebergSource implements DataSourceRegister, SupportsCatalogOptions {
+public class IcebergSource
```

Comment:
Another comment, is it now multiple ways to configure properties (including https://github.com/apache/iceberg/pull/4011), it may be confusing to user.  Worth to add a documentation about it, listing the precedence, ie:

I guess using dataframe API (to be double-checked)

- explicit dataframe option
- dataframe session default
- if table exists, explicit table option
- if table exists, table default

Reply:
I updated the docs and hope it's clear now.